### PR TITLE
Give the user the ability to opt out of 3rd party network validation when adding a new network

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

This PR gives the user control over the privacy settings for enabling or disabling a call to `https://chainid.network/chains.json`.

This is achieved by adding the toggle to the preferences controller and the condition in the appropriate place.

Integration tests are included.

* Fixes [#876](https://github.com/MetaMask/MetaMask-planning/issues/876)

## Screenshots/Screencaps

[insert screenshot from test]

## Manual Testing Steps

- Go to the privacy settings screen
- Disable validation
- Validate (in the network tab on dev tools) that the request is not made.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -5286,6 +5286,17 @@
   "usePhishingDetectionDescription": {
     "message": "Display a warning for phishing domains targeting Ethereum users"
   },
+  "useSafeChainsListValidation": {
+    "message": "Network details check"
+  },
+  "useSafeChainsListValidationDescription1": {
+    "message": "MetaMask uses a third-party service called",
+    "description": "useSafeChainsListValidationDescription1 is separated from useSafeChainsListValidationDescription2 so that we can bold the third party service name in the middle of them"
```
[the remark this answers is not in the document]
Comment (pedronfigueiredo) [reply app/_locales/en/messages.json:5294]: Thanks for the suggestion, @legobeat. I've looked at a few examples, but these seem to typically be used when a variable needs to be embedded in an internationalized string. In this case however, I want one internationalized string to be embedded in two others with a different style ([see](https://github.com/MetaMask/metamask-extension/pull/20816/files#diff-c67901cfc8925197a57b508a3cbe4601747c49fe78230721d405430c47db0a4dR355)). Am I missing something?

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -5286,6 +5286,17 @@
   "usePhishingDetectionDescription": {
     "message": "Display a warning for phishing domains targeting Ethereum users"
   },
+  "useSafeChainsListValidation": {
+    "message": "Network details check"
```
[the remark this answers is not in the document]
Comment (pedronfigueiredo) [reply app/_locales/en/messages.json:5290]: Thanks for the suggestion but I'd rather stick with the design copy -- [see here](https://www.figma.com/file/Ek7QDiB3TgvDhjPtmh1zAT/Privacy-Epic-Security%26Privacy-settings?type=design&node-id=387%3A16178&mode=design&t=V3Srh5x7hNWm0h7H-1).

File: test/e2e/tests/add-custom-network.spec.js [hunk 3]
```diff
@@ -56,68 +113,69 @@ describe('Custom network', function () {
...
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .build(),
+          title: this.test.title,
+        },
+        async ({ driver }) => {
+          await driver.navigate();
+          await driver.fill('#password', 'correct horse battery staple');
+          await driver.press('#password', driver.Key.ENTER);
```
Comment (DDDDDanica) [inline test/e2e/tests/add-custom-network.spec.js:175]:
```suggestion
        await unlockWallet(driver, WALLET_PASSWORD);
```

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 4]
```diff
@@ -340,6 +342,38 @@ export default class SecurityTab extends PureComponent {
...
+    const { useSafeChainsListValidation, setUseSafeChainsListValidation } =
+      this.props;
+
+    return (
+      <div ref={this.settingsRefs[2]} className="settings-page__content-row">
+        <div className="settings-page__content-item">
+          <span>{t('useSafeChainsListValidation')}</span>
+          <div className="settings-page__content-description">
+            {t('useSafeChainsListValidationDescription1') + ' '}
+            <b>chainid.network</b>
```
Comment (DDDDDanica) [inline ui/pages/settings/security-tab/security-tab.component.js:356]: This should be in the language file ~

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 5]
```diff
@@ -340,6 +342,38 @@ export default class SecurityTab extends PureComponent {
...
+            {' ' + t('useSafeChainsListValidationDescription2')}
+          </div>
+        </div>
+        <div className="settings-page__content-item">
```
Comment (DDDDDanica) [inline ui/pages/settings/security-tab/security-tab.component.js:360]: This class can be removed :) 
You can find examples here
https://github.com/MetaMask/metamask-extension/blob/06ff71a093d921d854d7d1b3a0a6baaff5f49e3e/ui/pages/settings/security-tab/security-tab.component.js#L213

File: test/e2e/helpers.js [hunk 6]
```diff
@@ -702,10 +692,25 @@ async function getEventPayloads(driver, mockedEndpoints, hasRequest = true) {
...
+
+/**
+ * When mocking the segment server and returning an array of mocks from the
+ * mockServer method, this method will allow getting all of the seen requests
+ * for each mock in the array.
+ *
+ * @param {WebDriver} driver
+ * @param {import('mockttp').Mockttp} mockedEndpoints
+ * @param {boolean} hasRequest
+ * @returns {import('mockttp/dist/pluggable-admin').MockttpClientResponse[]}
```
[the remark this answers is not in the document]
Comment (pedronfigueiredo) [reply test/e2e/helpers.js:710]: Interesting, but not that I'm aware of

**Review by DDDDDanica — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Review by albertolive — APPROVED** [review]

**Pull request merged**